File: transaction-service/src/main/java/ua/com/foxminded/restClient/exceptions/PersonNotFoundException.java
package ua.com.foxminded.restClient.exceptions;

import java.util.UUID;

public class PersonNotFoundException extends RuntimeException {

    static private final String message = "Person with id=%s not found.";

    public PersonNotFoundException() {
        super();
    }

    public PersonNotFoundException(String s, Throwable cause) {
        super(s, cause);
    }

    public PersonNotFoundException(Throwable cause) {
        super(cause);
    }

    public PersonNotFoundException(String s) {
        super(s);
    }

    public PersonNotFoundException(UUID uuid) {
        super(message.formatted(uuid));
    }

}

File: transaction-service/src/main/kotlin/ua/com/foxminded/restClient/exceptions/PersonNotFoundException.kt
package ua.com.foxminded.restClient.exceptions

import java.util.UUID

class PersonNotFoundException : RuntimeException {
    constructor() : super()
    constructor(s: String?, cause: Throwable?) : super(s, cause)
    constructor(cause: Throwable?) : super(cause)
    constructor(s: String?) : super(s)
    constructor(uuid: UUID?) : super(Companion.message.format(uuid))

    companion object {
        private const val message = "Person with id=%s not found."
    }
}
